### PR TITLE
Fixes #14277 - Added description field to users

### PR DESCRIPTION
**Pull request by dhlavac**

### CODE REVIEW

**Comment by ares** [comment]
[test]

File: app/controllers/api/v2/users_controller.rb [hunk 1]
```diff
@@ -92,6 +93,8 @@ def destroy
         end
       end
 
+
```
Comment (dLobatog) [inline app/controllers/api/v2/users_controller.rb]: Please remove these :smile:

Comment (dhlavac) [inline app/controllers/api/v2/users_controller.rb]: Ok sorry =D

**Comment by dLobatog** [comment]
This doesn't do what the ticket says (Customer want to store information in that description filed for users which is not alterable by any user other than org admin), however I'd argue we shouldn't have to implement this in Foreman. It was requested in 2010 by someone who used Spacewalk.. so I'm not even sure they want this anymore.

Apparently this field is wanted to provide random instructions to users.. which is not this software's goal.

cc @ares let me know what you think,  but IMO there's no need to make users any more complex than what they are now, especially when the original ticket wasn't even for us and is probably not relevant (6 years w/o this fix!)

**Comment by ares** [comment]
@dLobatog redmine issue says "Add a description field to users" which I find useful regardless of who can modify it. I don't think it makes the object more complex since there's no logic in this field. 

Regrading the BZ request, I'd argue that admin can set permissions in a way that only some roles can modify user objects. I know they wanted to restrict the specific field only but I don't think that's a good idea so I think this is a good compromise.

**Comment by ares** [comment]
[test]

**Comment by dLobatog** [comment]
@ares What is this useful for?

**Comment by ares** [comment]
You can add comments to it, you can use it for your organization custom information, identifiers or gecos. Since it's accessible via API, one can implement own workflows such as account expiration using the field. And it's searchable, so you can easily find users based on this custom information.

**Comment by dLobatog** [comment]
You could argue that for literally any object in Foreman, anything could have a description and it'd enable new workflows and possibly you can fill it with good information. That's just sidetracking from Foreman's purpose IMO - provisioning templates, locations, etc.. could have all a description field but it's just clutter in the majority of cases - in the case of users it's even more blatant as LDAP is where this kind of things go

**Comment by ares** [comment]
I would not oppose to add description field to most of Foreman objects. We have comment for host object already, how does that differ? We have audit comments. I think adding descriptions to locations and provisioning templates would be pretty useful even during development and debugging. How would my own description for a provisioning template "testing version for new networking" cause any clutter? I end up writing these comments in templates as comments. Actually we [ended up](https://github.com/theforeman/community-templates/blob/develop/pxe/PXELinux_local.erb#L6) doing it even in those we ship.

I understand that for simple objects like architecture it does not bring much but I still don't understand why adding description field to user is considered as sidetracking. If the argument is that users should use LDAP because it can address this use-case we could remove internal users completely or at least delete email field, password, first name and surname.

**Comment by dhlavac** [comment]
Is there something that can I change or improve ?

**Comment by ares** [comment]
[test]

**Comment by ares** [comment]
@dLobatog do you want me to ask for other reviewers opinions or can I merge this? (pending jenkins ofc)

**Comment by ares** [comment]
@dhlavac I think you need to add the field to rabl template too so it's available through API (show action)

**Comment by dhlavac** [comment]
@ares  added

**Comment by ares** [comment]
Thanks @dhlavac, merged as 6cc6c9ec41f8c3bccd263e634aa88a44484e5ec9. Could you please also create a simple patch for [hammer-cli-foreman](https://github.com/theforeman/hammer-cli-foreman/blob/master/lib/hammer_cli_foreman/user.rb#L33) so it starts displaying the field?

**Pull request closed without merging**